request.go

```diff
@@ -146,12 +149,26 @@ func (client *Client) doRequestWithRetries(req *http.Request, maxTime time.Durat
 		err  error
 		resp *http.Response
 		bo   = backoff.NewExponentialBackOff()
+		body []byte
 	)
 	bo.MaxElapsedTime = maxTime
 
+	// Save the body for retries
+	if req.Body != nil {
+		body, err = ioutil.ReadAll(req.Body)
+		if err != nil {
+			return resp, err
+		}
+	}
+
 	err = backoff.Retry(func() error {
+		r := bytes.NewReader(body)
+		req.Body = ioutil.NopCloser(r)
 		resp, err = client.HttpClient.Do(req)
 		if err != nil {
+			// This will leak datadog credentials due to the error containing
```

Comment:
Maybe put this information (plus related issue) a little higher so we don't need to duplicate it?